Update license-and-subscription.rst (#3672)
mattermost/docs: source/overview/license-and-subscription.rst

You can select **Please renew** to begin the renewal process. You can also select the **x** to dismiss the notification. The notification is reactivated when your browser refreshes.

How do I renew my license?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

You can renew your license in Mattermost via the blue renewal reminder banner. Select **Please Renew** to begin the process. You can also visit https://mattermost.com/renew, and complete the form provided. 

How long does it take to renew a license?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Once you’ve started the renewal process, we will be in contact with you to confirm your order and send you the order form. There may be additional paperwork required. Once we have the signed order form and (if applicable) the necessary paperwork from you, we are able to process the renewal and issue your license key within 24 hours.  

What happens to my trial/subscription if I don't renew in time?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you don't renew within the 60-day renewal period, a 10 day grace period is provided for you to upload a new license key. During this period your Mattermost installation runs as normal, with full access to Enterprise features. 

Author: Agreed. Change made.

When the grace period expires, your Enterprise version is downgraded to Team Edition. Enterprise features are disabled.
 
What happens when my license expires?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

If you don't renew your license within the 10-day grace period, your Mattermost version is automatically downgraded to Team Edition so you can still access and use Mattermost. However, Enterprise features will no longer be available and if you are currently using them, the functionality will no longer be accessible. 

When you renew, the Enterprise features will become available with the previous configuration (provided no action such as user migration has been taken). 

Which features are affected when my Enterprise license expires?
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''

The affected Enterprise features include, but are not limited to, the following:

.. csv-table::

>     :header: "Feature", "How it's affected", "What steps do I need to take?"

Reviewer: I'm unsure if the items on the LHS of the table need periods.

Author: Good point. I've removed them, looks better. :)

    "Elasticsearch.", "Elasticsearch is automatically disabled and will start using the default database for indexing posts.", "None needed."

>     "AD/LDAP, SAML SSO, Office 365 SSO, and Google SSO login options removed from the sign-in page.", "Users who previously signed in with one of these methods are no longer able to. 

Author: Great suggestion, thank you! I've updated the table.

    "AD/LDAP.", "Groups in the database are retained but cannot be used. Memberships are frozen in state for group synced teams/channels. 
    
    Mentions for AD/LDAP groups are not shown in the autocomplete menu. 
    
    Group mentions are no longer highlighted in text and do not trigger new notifications.", "Use the `CLI <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-group>`_ to modify group sync settings for the team/channel."
    "High Availability.", "High Availability is disabled. If all nodes in a cluster continue running, the nodes will stop communicating and caches will get out of sync. This is likely to cause delays in messages, notifications, etc.", "None needed."
    "Performance monitoring.", "Monitoring is disabled and Grafana will no longer update with new data.", "None needed."
    "Compliance exports.", "Jobs are no longer scheduled in the job server. Data is not exported.", "None needed."
    "Data retention.", "Jobs are no longer scheduled in the job server. Data is not deleted.", "None needed."

>     "Custom terms.", "Custom terms no longer displayed to end users on login. Data is retained in the Terms of Service database table."

Reviewer 2: ```suggestion
    "Custom terms.", "Custom terms no longer displayed to end users on login. Data is retained in the Terms of Service database table.", "None needed."
```

    "Custom announcement banners are no longer visible.", "Default announcement banner displayed.", "None needed."
    "Multi-factor authentication (MFA).", "MFA is no longer enforced/required for new accounts but remains enabled for those who configured it.", "None needed."
    "Permissions.", "Permissions are retained in the database in a frozen state and cannot be modified in the the System Console.", "Use the `CLI <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-permissions-reset>`_ to reset permissions to default."
    "Guest accounts.", "Guests that are not actively logged in are prevented from logging in. Guests who are actively logged in are able to use Mattermost until their session expires or they log out.", "None needed."

>     

Author: I've edited the first line to read: "The affected Enterprise features include, but are not limited to, the following:"

Yes, once your payment is successfully processed your license is immediately available to download.

> How will I know when to renew my license?

Reviewer 2: Let's also add an FAQ on "What happens when trial or subscription expires?"

Yes, once your payment is successfully processed your license is immediately available to download.

How will I know when to renew my license?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

You will be notified 60 days prior to your license expiry that your license is due for renewal, via a blue banner displayed at the top of your Mattermost window. This banner is only visible to System Admins.

You can select **Please renew** to begin the renewal process. You can also select the **x** to dismiss the notification. The notification is reactivated when your browser refreshes.

How do I renew my license?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> You can renew your license in Mattermost via the blue renewal reminder banner. Select Renew Now to begin the process. You can also visit https://mattermost.com/renew, and complete the form provided. 

Reviewer 2: ```suggestion
You can renew your license in Mattermost via the blue renewal reminder banner. Select **Please Renew** to begin the process. You can also visit https://mattermost.com/renew, and complete the form provided. 
```

How long does it take to renew a license?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Once you’ve started the renewal process, we will be in contact with you to confirm your order and send you the order form. There may be additional paperwork required. Once we have the signed order form and (if applicable) the necessary paperwork from you, we are able to process the renewal and issue your license key within 24 hours.  

What happens to my trial/subscription if I don't renew in time?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> From version 5.24, if you don't renew within the 60-day renewal period, a 10 day grace period is provided. During this period your Mattermost installation runs as normal, with full access to Enterprise features. 

Reviewer 2: ```suggestion
If you don't renew within the 60-day renewal period, a 10 day grace period is provided for you to upload a new license key. During this period your Mattermost installation runs as normal, with full access to Enterprise features. 
```

When the grace period expires, your Enterprise version is downgraded to Team Edition. Enterprise features are disabled.
 
What happens when my trial/subscription expires?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

If you don't renew your license within the 10-day grace period, your Mattermost version is automatically downgraded to Team Edition so you can still access and use Mattermost. However, Enterprise features will no longer be available and if you are currently using them, the functionality will no longer be accessible. 

When you renew, the Enterprise features will become available with the previous configuration (provided no action such as user migration has been taken). 

> Which features are affected when my Enterprise license expires?

Reviewer 2: I'm wondering if this is part of the previous question above, "What happens when my trial/subscription expires?"

Author: It was. They are similar questions but one is focused more on the mechanics of the expiry and one on the features affected. I'm open to suggestions - wondering if maybe the questions are what needs work.

Reviewer 2: True. It could be a subsection of the FAQ entry above it. 0/5 though

Author: I like the idea of a subsection. That'll work.

Author: Subsection added.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you are using the following Enterprise features, they will be impacted 

Reviewer: ```suggestion
If you are using the following Enterprise features, they will be impacted.
```

.. csv-table::
    :header: "Feature", "How it's affected", "What steps do I need to take?"

>     "Elasticsearch", "Elasticsearch is automatically disabled and will start using the default database for indexing posts", "None needed."

Reviewer: Some of the items in the LHS column have periods, some don't.

    "AD/LDAP, SAML SSO, Office 365 SSO, and Google SSO login options removed from the sign-in page.", "- Users who previously signed in with one of these methods are no longer able to. 

>     - Users who were already signed in can continue to use Mattermost until their session expires or until they log out.", "Users must be migrated to email authentication using the `CLI <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-user-migrate-auth>`_. This can also be done via **System Console > Users**. Select the drop-down menu for the relevant member, choose **Switch to Email/Password**, enter a new password, and choose **Reset**."

Reviewer 2: For consistency, propose not using bullets in this case, and instead using paragraphs separated by a blank line.

![image](https://user-images.githubusercontent.com/13119842/84180481-8cd60600-aa55-11ea-8c92-fe3bfcd04c4f.png)

Author: Good idea. Updated. :)

>     - Users who were already signed in can continue to use Mattermost until their session expires or until they log out.", "Users must be migrated to email authentication using the `CLI <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-user-migrate-auth>`_. This can also be done via **System Console > Users**. Select the drop-down menu for the relevant member, choose **Switch to Email/Password**, enter a new password, and choose **Reset**."

Reviewer 2: Looks like the CLI doesn't support migrating "to" email. So the CLI option isn't actually viable.

    "AD/LDAP groups in the database are retained but cannot be used", "- Memberships are frozen in state for group synced teams/channels. 
    - Mentions for AD/LDAP groups are not shown in the autocomplete menu. 
    - Group mentions are no longer highlighted in text and do not trigger new notifications.", "Use the `CLI <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-group>`_ to modify group sync settings for the team/channel."

>     "High availability is disabled.", "If all nodes in a cluster continue running, the nodes will stop communicating and caches will get out of sync. This is likely to cause delays in messages, notifications, etc.", "None needed."

Reviewer 2: "If all nodes in a cluster continue running..." --> does this indicate the admin does need to take action so that the nodes are no longer been used?

Author: I couldn't find anything in [this guide](https://docs.mattermost.com/deployment/cluster.html) about shutting down/discontinuing a cluster other than for maintenance. I'm not certain which of the steps would apply in this instance.

Reviewer 2: @justinegeffen we could check in with the Enterprise team to confirm

Author: Asked them here: https://community.mattermost.com/core/pl/j9rza84tipy1tk7yc3bn1ho8qy.

    "Performance monitoring is disabled.", "Grafana will no longer update with new data.", "None needed."
    "Compliance exports jobs are no longer scheduled in the job server", "Data is not exported.", "None needed."
    "Data retention jobs are no longer scheduled in the job server", "Data is not deleted.", "None needed."

>     "Custom terms no longer show for end users on login", "Data is retained in the Terms of Service database."

Reviewer 2: ```suggestion
    "Custom terms no longer show for end users on login", "Data is retained in the Terms of Service database table."
```

    "Custom announcement banners are no longer visible.", "Default announcement banner displayed.", "None needed."
    "Multi-factor authentication (MFA)", "MFA is no longer enforced/required for new accounts but remains enabled for those who configured it.", "None needed."
    "Permissions", "Permissions are retained in the database in a frozen state and cannot be modified in the the System Console.", "Use the `CLI <https://docs.mattermost.com/administration/command-line-tools.html#mattermost-permissions-reset>`_ to reset permissions to default."
    "Guest accounts", "Guests that are not actively logged in are prevented from logging in. Guests who are actively logged in are able to use Mattermost until their session expires or they log out.", "None needed."
    
Why can't I dismiss the expiry notification banner?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If there is a red expiry announcement banner stating: "Enterprise license is expired and some features may be disabled. Please contact your System Administrator for details." it means your grace period has expired. This announcement banner persists until the license is renewed, and is visible to users as well as System Admins.

Reviewer 2: ```suggestion
If there is a red expiry announcement banner stating: "Enterprise license is expired and some features may be disabled. Please contact your System Administrator for details." it means your grace period has expired. This announcement banner persists until the license is renewed, and is visible to users.
```

Once a new license is applied, the banner will no longer be visible. 

> If you'd prefer to use Mattermost Team Edition going forward, you need to revoke the expired license by following these steps. 

Reviewer: Maybe missing a link in "these steps".

Author: Good catch, thanks! I had meant to add those.

> If you'd prefer to use Mattermost Team Edition going forward, you need to revoke the expired license by following these steps. 

Reviewer 2: ```suggestion
If you do not plan to renew your Enterprise Edition subscription, revoke the expired license in **System Console > Edition and License**.
```
